modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/KleisliInstances.kt

```diff
@@ -0,0 +1,30 @@
+package arrow.effects
+
+import arrow.Kind
+import arrow.core.Either
+import arrow.data.Kleisli
+import arrow.data.KleisliPartialOf
+import arrow.data.fix
+import arrow.effects.typeclasses.MonadDefer
+import arrow.instance
+
+
+@instance(Kleisli::class)
+interface KleisliMonadDeferInstance<F, D>: MonadDefer<KleisliPartialOf<F, D>> {
+
+    fun FF(): MonadDefer<F>
+
+    override fun <A> defer(fa: () -> Kind<KleisliPartialOf<F, D>, A>): Kind<KleisliPartialOf<F, D>, A> = Kleisli { d: D ->
```

Comment:
Can you please move this constructor to the Kleisli file instead? That way everyone can use it.

Reply:
I think it is not possible to move it to Kleisli because of the dependency on MonadDefer